### PR TITLE
make it clear that maven_artifactId==bazel_target_name

### PR DESCRIPTION
**Pull request by plaird**

### CODE REVIEW

File: maven/maven_functions.sh [hunk 1]
```diff
@@ -86,7 +86,7 @@ _for_each_pom() {
             # the filename of the jar built by Bazel uses this pattern:
             jar_artifact_path="$build_dir_package_path/lib${package_name}.jar"
             if [ ! -f "${jar_artifact_path}" ]; then
-                echo "WARN: lib${package_name}.jar not found, looking for alternatives"
+                echo "WARN: lib${package_name}.jar not found at path ${jar_artifact_path}, looking for alternatives..."
```
Comment (plaird) [inline maven/maven_functions.sh:89]: having the path in the log message help me isolate the issue we had internally

**Comment by plaird** [comment]
@simontoens another idea would be to fail during the alternate jar search if the found jar file is below 500 bytes since that is probably an empty jar file with only a MANIFEST.MF

**Comment by simontoens** [comment]
@plaird why do we actually end up with empty jars?

**Review by simontoens — APPROVED** [review]

**Comment by plaird** [comment]
It is a good question. Bazel is creating them for the java_binary rule. It is the name of the rule without the _deploy suffix.

 jar -tvf  bazel-bin/examples/hello-world/juicer/make-juice.jar
   0 Fri Jan 01 00:00:00 MST 2010 META-INF/
   100 Fri Jan 01 00:00:00 MST 2010 META-INF/MANIFEST.MF

So if you are trying to deploy the java_binary _deploy.jar, and you forget to run the :* target, this logic will pick the make-juice.jar because make-juice_deploy.jar is not there.

**Comment by simontoens** [comment]
Or we reverse the order?  First, look for the _deploy jar?

**Comment by plaird** [comment]
But we may also want a failure check if the jar -tvf lists just the MANIFEST.MF as I don't think that is ever wanted for upload.

**Comment by simontoens** [comment]
Yes I can't think of a reason why we would want to upload a jar that only has MANIFEST.MF.  Are you going to make these changes?

**Comment by plaird** [comment]
We actually already check for xyz_deploy.jar before xyz.jar. The extra feature here would be the check for a MANIFEST only jar.

**Pull request merged**